### DISCUSSION TITLE
I need help running stellarium on my Raspberry pi 4b

### DISCUSSION THREAD

**Discussion by farm24** [Stellarium/stellarium#1846] [Q&A]
When I run it it just has a tiny window that is frozen
![49535C96-DD5D-402A-8DB2-C8F502BC39C8](https://user-images.githubusercontent.com/77849362/130363987-b071c133-4a09-4af9-8097-c48e879e3fdc.jpeg)
See picture

**Comment by gzotti** [comment]
https://github.com/Stellarium/stellarium/wiki/FAQ#reporting-a-crash

**Reply by farm24** [reply]
I’ll just say here that I’m running the latest rasbian and it’s off of snapd

**Reply by gzotti** [reply]
I cannot comment on snapd. I just updated my 4B and rebuilt following our own experience noted down in https://github.com/Stellarium/stellarium/wiki/Raspberry-Pi

**Reply by farm24** [reply]
Where can I find how to build it?

**Reply by alex-w** [reply]
> Where can I find how to build it?

Previously link + https://github.com/Stellarium/stellarium/blob/master/BUILDING.md

**Reply by gzotti** [reply]
Building instructions are conventionally found in BUILDING.md in the source tree.

**Reply by farm24** [reply]
I can’t find OpenGL just gl driver and I can’t find full kms

**Reply by gzotti** [reply]
Would you mind reading to the end?

**Reply by farm24** [reply]
Ok I read and I’m compiling but it keeps saying Cmake error: the source directory “/home/pi” does not appear to contain CMakeLists.txt

**Reply by gzotti** [reply]
Are you familiar with compiling software or is this your first time experience?

**Reply by farm24** [reply]
First time as I am just trying to get a portable stellarium station up

**Reply by gzotti** [reply]
OK. Let's combine instructions from both given sources: Open a terminal.
Install dependencies:
$ sudo apt install build-essential cmake zlib1g-dev libgl1-mesa-dev libdrm-dev gcc g++ \
                 graphviz doxygen gettext git \
                 qtbase5-dev qtscript5-dev libqt5svg5-dev qttools5-dev-tools qttools5-dev libqt5opengl5-dev \
                 qtmultimedia5-dev libqt5multimedia5-plugins libqt5serialport5 libqt5serialport5-dev qtpositioning5-dev \
                 libgps-dev libqt5positioning5 libqt5positioning5-plugins
$ mkdir DEV
$ cd DEV
$ git clone --filter=blob:none https://github.com/Stellarium/stellarium.git
$ mkdir build; cd build
$ cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX=/opt/stellarium ../stellarium
$ nice make -j4 

Take a coffee or two. After building, go to the source directory and test:

$ cd ../stellarium
$ ../build/src/stellarium

or, install. This should be as easy as 
$ cd ../build
$ sudo make install

(But I have not tried that today)

**Reply by farm24** [reply]
i did the first one on the bottom and it didn't work but i managed to get it to work by using ../DEV/build/src/stellarium thank you for making this little guide!

**Comment by t4saha** [comment]
In general snap package has some issue with opengl on RPi4 currently.
Please try edge channel `sudo snap install stellarium-daily --edge`
